src/components/app/ProfileSharing.js

```diff
@@ -262,20 +295,43 @@ class ProfileSharingCompositeButton extends PureComponent<
     });
   }
 
+  _onSecondSharePanelOpen() {
+    const { profileSharingStatus } = this.props;
+    this.setState({
+      shareNetworkUrls: !profileSharingStatus.sharedWithUrls,
+    });
+  }
+
   render() {
     const { state, uploadProgress, error, shortUrl } = this.state;
-    const { symbolicationStatus } = this.props;
+    const { profile, symbolicationStatus, profileSharingStatus } = this.props;
+
     const shareLabel =
       symbolicationStatus === 'DONE'
         ? 'Share...'
         : 'Sharing will be enabled once symbolication is complete';
+
+    const secondShareButtonVisibility =
+      // It must be shared before
+      state === 'public' &&
+      // If both with and without URL is shared, do not show anymore.
+      (!profileSharingStatus.sharedWithUrls ||
+        !profileSharingStatus.sharedWithoutUrls) &&
+      // If profile is shared without network URLs and they are removed from the profile, do not show.
+      (profileSharingStatus.sharedWithUrls || !profile.meta.networkURLsRemoved);
```

Comment:
I'd very much like to avoid the need for this new meta. But I think it's needed if we want to be able to load a profile that's been shared with URLs, and share it again without URLs.

Now is it something that we want? I honestly didn't think of this case before. I assumed that if we loaded a profile from the public storage, we never wanted to share it again. But I see your point. What do you think?

Reply:
The main reason for that flag is to hide the share with URLs button when there is no URL information in the loaded public profile.

We can also say that if the loaded profile is from public, then do not show any share button again. I haven't thought that until you comment, actually. I'm okay with both but I think being able to sharing it again would be nice(or confusing for the user?).

Reply:
I wouldn't have done it in the first place but because now the code is here let's keep it. I think this is still a valuable addition :)